python/ray/train/_internal/state/state_manager.py

```diff
@@ -82,14 +87,38 @@ def collect_train_worker_info():
             for ds_name, ds in datasets.items()
         ]
 
-        train_run_info = TrainRunInfo(
+        updates = dict(
             id=run_id,
             job_id=job_id,
             name=run_name,
             controller_actor_id=controller_actor_id,
             workers=worker_info_list,
             datasets=dataset_info_list,
             start_time_ms=start_time_ms,
+            run_status=run_status,
+            status_detail=status_detail,
         )
 
-        ray.get(self.state_actor.register_train_run.remote(train_run_info))
+        # Clear the cached info to avoid registering the same run twice
+        self.train_run_info_dict[run_id].clear()
```

Comment:
Will this populate the dictionary on the first call? Since the first line in `_update_train_run_info` will check if the `run_id` is present.

Reply:
Yes it will populate the dict. But yeah I think it's a bit implicit, let me update it to

`self.train_run_info_dict[run_id] = {}`